aws_advanced_python_wrapper/limitless_connection_plugin.py

```diff
@@ -0,0 +1,543 @@
+                except Exception:
+                    # Ignore
+                    pass
+
+            if context is not None:
+                context.close_context()
+
+    def _open_connection(self):
+        try:
+            driver_dialect = self._plugin_service.driver_dialect
+            if self._monitoring_conn is None or driver_dialect.is_closed(self._monitoring_conn):
+                logger.debug("LimitlessRouterMonitor.OpeningConnection", self.host_info.url)
+                self._monitoring_conn = self._plugin_service.force_connect(self._host_info, self._properties, None)
+                logger.debug("LimitlessRouterMonitor.OpenedConnection", self._monitoring_conn)
+
+        except Exception:
+            if self._monitoring_conn is not None:
+                try:
+                    self._monitoring_conn.close()
+                except Exception:
+                    pass  # ignore
+
+                    self._monitoring_conn = None
+
+
+class LimitlessQueryHelper:
+
+    _DEFAULT_QUERY_TIMEOUT_MS: int = 5000
+
+    def __init__(self, plugin_service: PluginService):
+        self._plugin_service = plugin_service
+
+    def query_for_limitless_routers(self, connection: Connection, host_post_to_map: int) -> List[HostInfo]:
+
+        database_dialect = self._plugin_service.database_dialect
+        if not isinstance(database_dialect, AuroraLimitlessDialect):
+            raise UnsupportedOperationError(Messages.get("LimitlessQueryHelper.UnsupportedDialectOrDatabase"))
+        aurora_limitless_dialect: AuroraLimitlessDialect = database_dialect
+        query = aurora_limitless_dialect.limitless_router_endpoint_query
+
+        with closing(connection.cursor()) as cursor:
+            self._plugin_service.driver_dialect.execute("Cursor.execute",
+                                                        lambda: cursor.execute(query),
+                                                        query,
+                                                        exec_timeout=LimitlessQueryHelper._DEFAULT_QUERY_TIMEOUT_MS)
+            return self._map_result_set_to_host_info_list(cursor.fetchall(), host_post_to_map)
+
+    def _map_result_set_to_host_info_list(self, result_set: List[Tuple[Any, Any]], host_post_to_map: int) -> List[HostInfo]:
+        print(result_set)
+        list_of_host_infos: List[HostInfo] = []
+        for result in result_set:
+            list_of_host_infos.append(self._create_host_info(result, host_post_to_map))
+        return list_of_host_infos
+
+    def _create_host_info(self, result: Tuple[Any, Any], host_post_to_map: int) -> HostInfo:
+        host_name: str = result[0]
+        cpu: float = float(result[1])
+
+        weight: int = round(10 - (cpu * 10))
+        if weight < 1 or weight > 10:
+            weight = 1
+            logger.debug("LimitlessRouterMonitor.InvalidRouterLoad", host_name, cpu)
+
+        return HostInfo(host_name, host_post_to_map, weight=weight, host_id=host_name)
```

Comment:
In JDBC we also set the role to writer and availability to AVAILABLE, should we do that here?